app/presenters/setting_presenter.rb

```diff
@@ -7,38 +7,32 @@ class SettingPresenter
   attribute :id, :integer
   attribute :category, :string, default: 'Setting::General'
   attribute :name, :string
-  attribute :default
   attribute :value
+  attribute :default
   attribute :description, :string
   attribute :full_name, :string
   attribute :encrypted, :boolean, :default => false
   attribute :settings_type, :string
-  attribute :select_values
-  attribute :config_file
-  attribute :updated_at, :datetime
-  attribute :created_at, :datetime
-  attr_accessor :options
-
-  def self.from_setting(setting)
-    SettingPresenter.new({id: setting.id,
-                          name: setting.name,
-                          category: setting.category,
-                          description: setting.description,
-                          settings_type: setting.settings_type,
-                          default: setting.default,
-                          full_name: setting.full_name,
-                          updated_at: setting.updated_at,
-                          created_at: setting.created_at,
-                          config_file: setting.class.config_file,
-                          select_values: setting.select_collection,
-                          value: setting.value,
-                          encrypted: setting.encrypted? })
-  end
+  attribute :config_file, :string
+
+  delegate :updated_at, :errors, to: :db_model
+
+  attr_reader :db_model
 
   def self.model_name
     Setting.model_name
   end
 
+  # This refreshes a value for the entry from db
+  def load_db_value(db_model)
+    @db_model = db_model
+    if db_model
+      v = db_model[:value]
+      v = db_model.decrypt_field(v)
+    end
+    self.value = v.nil? ? default : YAML.safe_load(v)
+  end
```

Comment:
Thinking more about this... we could keep it only in the model and set the value directly according to what we get from the model. It might be that it's the default, but that's ok. It would make loading simpler as well (see next comment)

Reply:
yeah, make sense, the serialization should be left in the data layer, so it will be much cleaner :+1: